src/recipes.jl

```diff
@@ -145,9 +143,9 @@ end
 @deps hspan shape
 
 @recipe function f(::Type{Val{:vspan}}, x, y, z)
-    n = div(length(y), 2)
-    newx = vcat([[y[2i - 1], y[2i - 1], y[2i], y[2i], NaN] for i = 1:n]...)
-    newy = repeat([-Inf, Inf, Inf, -Inf, NaN], outer = n)
+    n = div(length(x), 2)
+    newx = vcat([[x[2i - 1], x[2i - 1], x[2i], x[2i], NaN] for i = 1:n]...)
+    newy = repeat([-Inf, Inf, Inf, -Inf, NaN], outer=div(length(x),2))
```

Comment:
For consistency I would move this also to the shorthands section and pass `y`

Reply:
I have done this together with my fix with macro mentioned above but forgot to commit this bit separately... I hope that's OK

Reply:
So I reverted this because somehow the `Inf`s are turned into `NaN`s here. (This is why the `vspan` plot was not showing anything above.)